Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift

```diff
@@ -0,0 +1,223 @@
+//
+// Copyright (C) 2015-2019 Virgil Security Inc.
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     (1) Redistributions of source code must retain the above copyright
+//     notice, this list of conditions and the following disclaimer.
+//
+//     (2) Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in
+//     the documentation and/or other materials provided with the
+//     distribution.
+//
+//     (3) Neither the name of the copyright holder nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+//
+
+import VirgilSDK
+import VirgilCrypto
+
+// MARK: - Extension with peer-to-pear encrypt and decrypt operations
+extension EThree {
+    /// Signs then encrypts data (and signature) for user
+    ///
+    /// - Important: Deprecated decrypt method is unable to decrypt result of this method
+    ///
+    /// - Parameters:
+    ///   - data: data to encrypt
+    ///   - user: user Card to encrypt for
+    /// - Returns: encrypted data
+    /// - Important: Automatically includes self key to recipientsKeys.
+    /// - Important: Requires private key in local storage
+    @objc(signThenEncryptData:forUser:error:)
+    open func signThenEncrypt(data: Data, for user: Card) throws -> Data {
```

Comment:
1. Rename to authEncrypt
2. What's the point of setting explicit objc name?

Reply:
> What's the point of setting explicit objc name?
Conflicts of encrypt(for user: Card) and encrypt(for users: [Cards]). It has same signature in objc

Reply:
Renamed